frb/analysis/dmhalos_photoz.py

```diff
@@ -0,0 +1,615 @@
+import numpy as np, os, glob
+
+from astropy.io import fits
+from astropy.table import Table, vstack, join
+from astropy.coordinates import SkyCoord
+from astropy import units as u
+from astropy.cosmology import Planck15 as p15
+from astropy import visualization as vis
+from astropy.wcs import WCS
+from astropy.stats import sigma_clipped_stats
+
+from scipy.interpolate import interp1d, interp2d, RegularGridInterpolator
+from scipy.sparse import lil_matrix, save_npz
+from scipy import stats as sci_st
+from scipy.integrate import simps
+
+from frb.halos.models import ModifiedNFW, halomass_from_stellarmass
+from frb.frb import FRB
+from frb.galaxies import cigale as frbcig
+from frb.galaxies import eazy as frb_ez
+from frb.surveys import des
+from frb.surveys.catalog_utils import convert_mags_to_flux
+
+from pathos.multiprocessing import ProcessingPool as Pool
+import progressbar
+
+from matplotlib import pyplot as plt
+
+from threedhst import eazyPy as ez
+
+from importlib import reload
+
+DEFAULT_DATA_FOLDER = "data"
+
+def get_des_data(coords:SkyCoord, radius:u.Quantity=15.*u.arcmin, starbright:float=17,
+                 starflagval:float=0.9, gaiacat:str=None, write:bool=False, outfile:str=None)->Table:
+    """
+    Download photometry for galaxies within an FRB field.
+    Args:
+        coords (SkyCoord): Coordinates of the center of a cone search.
+        radius (Quantity, optional): Radius of cone search.
+        starbright (float, optional): Lower limit of r band mag. Objects brighter
+                                    than this will be removed.
+        starflagval (float, optional): Upper limit for a morphology-based classifier
+                                       flag. Objects more point-like (i.e. higher value)
+                                       will be filtered out.
+        gaicat (str, optional): Optional file with gaia catalog of stars within the same search
```

Comment:
Specify the format and columns of this table.

Reply:
Done.